components/content-service/pkg/archive/tar.go

```diff
@@ -69,9 +75,86 @@ func WithGIDMapping(mappings []IDMapping) BuildTarbalOption {
 	}
 }
 
+// ExtractTarbal extracts an OCI compatible tar file src to the folder dst, expecting the overlay whiteout format
+func ExtractTarbal(ctx context.Context, src io.Reader, dst string, opts ...TarOption) (err error) {
+	var cfg tarConfig
+	start := time.Now()
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	//nolint:staticcheck,ineffassign
+	span, ctx := opentracing.StartSpanFromContext(ctx, "extractTarbal")
+	span.LogKV("src", src, "dst", dst)
+	defer tracing.FinishSpan(span, &err)
+
+	re, wr := io.Pipe()
```

Comment:
super nit: more idiomatic name choice would be `pr, pw := io.Pipe()` or `p, w := io.Pipe()` (e.g. [here](https://golang.org/pkg/io/#Pipe))